bin/eslint.js

```diff
@@ -57,6 +57,21 @@ process.once("uncaughtException", err => {
     process.exitCode = 2;
 });
 
+/**
+ * Handles the condition based Promise as returned by cli.execute()
+ * @param {string | void} arg argument to be passed in
+ * @returns {void}
+ */
+function handleArgs(arg) {
+    const exitCode = cli.execute(process.argv, arg ? arg : null);
+
+    process.exitCode = exitCode instanceof Promise
```

Comment:
It looks like `process.exitCode` will be an unresolved promise when `cli.execute` returns a promise.

Reply:
The respective block checks whether `cli.execute()` Promise` was returned and waits till it resolves (IIFE).

Reply:
The respective block checks whether `cli.execute()` returns a `Promise` and waits until it gets resolved :thinking:

Reply:
It's `await`ing the resolution of the `Promise` within the IIFE, but the resolution value of the `Promise` returned by `exitCode` isn't being `await`ed. It's also not returning the resolved value of `exitCode`, so when it does resolve `process.exitCode` will be `undefined`.

An alternative would be 

```js
async function handleArgs(arg) {
  const exitCode = cli.execute(process.argv, arg ? arg : null);

  process.exitCode = exitCode instanceof Promise
    ? await exitCode
    : exitCode;
}
```

That being said, this is in the module scope, so we'd have to wrap all this logic in an `async` IIFE or have it use promises.